chart/templates/deployment.yaml

```diff
@@ -106,12 +106,12 @@ spec:
           periodSeconds: 10
         resources: {{- toYaml .Values.controller.manager.resources | nindent 10 }}
         securityContext:
-          allowPrivilegeEscalation: false
+          {{- toYaml .Values.controller.manager.securityContext | nindent 12 }}
         volumeMounts:
         - mountPath: /var/run/podinfo
           name: podinfo
       securityContext:
-        runAsNonRoot: true
+        {{- toYaml .Values.controller.podSecurityContext | nindent 8 }}
```

Comment:
Since this PR was opened, we've added a pre-delete hook that runs the same image as the manager container further down in this file. I think we should probably apply the `.Values.controller.podSecurityContext` and `.Values.controller.manager.securityContext` settings to the pod and container respectively in that job. WDYT?

Reply:
I pushed a commit that includes some test updates to better illustrate this suggestion: f6a5aa8. If your environment has a requirement for setting `securityContext` on all pods, I think it will be required.